app/controllers/application_controller.rb

```diff
@@ -20,4 +20,18 @@ def current_user
   def redirect_to_home
     redirect_to root_path
   end
+
+  def login_required(login_required_page)
+    return if current_user
+
+    session[:return_to] = request.fullpath
+
+    redirect_to login_path, alert: t('auth.login_required', page: login_required_page)
+  end
+
+  def redirect_to_target_or_default(default)
+    redirect_to(session[:return_to] || default)
+
+    session.delete(:return_to)
```

Comment:
should we move the `session.delete(:return_to)` above the `redirect_to(session[:return_to] || default)` 🤔 as I'm not 100 sure that action can perform after the `redirect action`

Reply:
I think we cannot move `session.delete(:return_to)` above the redirect because if we move above then `session[:return_to]` will always be nil and redirect_to will redirect to default

I have a test to make sure that `session[:return_to]` will get deleted after redirect.
https://github.com/gutakk/ruby-google-scraper/blob/509887251266283807b3940469ee4588ee29ec7f/spec/controllers/sessions_controller_spec.rb#L73-L80

What do you think?

Reply:
can we asssign the session to another variable

```
return_to_path = session.delete(:return_to)

redirect_to(return_to_path || default)
```

Reply:
@andyduong1920 @gutakk redirect_to won't stop the code execution, it just set the HTTP responses 
https://guides.rubyonrails.org/layouts_and_rendering.html#using-redirect-to
> redirect_to and redirect_back do not halt and return immediately from method execution, but simply set HTTP responses. Statements occurring after them in a method will be executed. You can halt by an explicit return or some other halting mechanism, if needed.

but I agree that with your suggestion that redirect_to should be the last statement to avoid the confusion

Reply:
@andyduong1920 @longnd Good catch. I will make the change

Reply:
Fixed in de552fd8394bebf66c9684760b3b7bf07207e9db